src/incubator/gradient/index.tsx

```diff
@@ -0,0 +1,38 @@
+import React, {useEffect} from 'react';
+import {LinearGradientPackage} from 'optionalDeps';
+const LinearGradient = LinearGradientPackage?.default;
+import {LogService} from 'services';
+import {GradientProps, RectangleGradientProps, CircleGradientProps, BorderGradientProps} from './types';
+import RectangleGradient from './RectangleGradient';
+import CircleGradient from './CircleGradient';
+import BorderGradient from './BorderGradient';
+
+export {GradientProps};
+
+const Gradient = (props: GradientProps) => {
+  const {type = 'rectangle', ...others} = props;
+
+  useEffect(() => {
+    if (LinearGradient === undefined) {
+      LogService.error(`RNUILib Gradient requires installing "react-native-linear-gradient" dependency`);
+    }
+  }, []);
+
+  if (!LinearGradient) {
```

Comment:
This if `(!LinearGradient)` check is irrelevant because `LinearGradient` cannot be falsy (it's either a component or a function).
The fallback function `() => null` will pass this check consider checking `if (LinearGradient === null)` instead.

Reply:
Changed a little